### ISSUE TITLE
Notiuni de baza Machine Learning (ML) si Neuronal Networrks (NN)

### ISSUE THREAD

**Issue by megadanyx** [megadanyx/Python-VN-HW-#3]
## Notiuni de baza Machine Learning (ML) si Neuronal Networrks (NN)

1. Ce metode de trainining exista in ML (Learning modes)?
2. Ce este o retea neuronala (NN)?
3. Ce reprezinta greutatea (weight)?
4. Ce inseamna "strat neuronal" (layer)?
5. Ce este o "epoca" (epoch)?
6. Ce reprezinta "functia de activare" (activation function)?
7. Cum poate fi crescuta acuratetea modelului antrenat (accuracy)?
8. Ce sa-r schimba daca ar fi marit numarul de neuroni de sute de ori, la fel si numarul de straturi cu neuroni ?
9. Cum mai este denumit trainingul / modelul cand se utilizeaza o retea cu mai multe straturi ?

**Comment by megadanyx** [comment]
### 1. Ce metode de trainining exista in ML (Learning modes)?
**Metodele se clasifica in 2 categorii ,supervised learning si unsupervised learning care se deosebesc prin modul de procesare a datelor ,structura prin care sunt create etc..  , sunt folosit in dependenta de domeniu in care se solicita neuronul....**

**What is supervised learning?**
The supervised learning technique is more commonly used in [machine learning](https://www.tableau.com/learn/articles/define-machine-learning) because it deals with straightforward tasks and is easy to implement. Data inputs are labeled with the answer that the algorithm should arrive at, which helps the machine pick out patterns in the future, better differentiate data, or make predictions. Supervised learning is classified into two categories of algorithms and is ideal for problems where there are reference points available.

**Classification:** A classification problem exists when the output variable is a specific category.
**Regression:** A regression problem exists when the output variable is a real value that fluctuates (i.e. dollars, weight, measurement).
We regularly use supervised learning to teach ourselves or someone else a new task. It’s kind of like being given a test with the answer key. Once you have the task mastered, this technique can be applied to similar processes and information.

**What is unsupervised learning?**
In this technique, the machine learning model learns organically instead of receiving a data set with explicit instructions. It then tries to automatically find structure in the raw data through analysis and interpretation. While supervised learning is easiest, we don’t always have access to complete, perfectly labeled data sets to train the algorithm. Where supervised learning has the “right” answer, unsupervised learning is helpful in situations where analysts (or really anyone) ask questions and the algorithm doesn’t have the answer, or there’s more than one answer. The unsupervised learning model is classified into four different categories of algorithms, which group data based on similarities or relationships among variables:

**Clustering:** The deep learning model looks for data and features that are similar then groups them together.
**Association:** By reviewing key attributes in the data, an unsupervised learning model can predict other attributes that they’re commonly associated with.
**Anomaly detection:** In this instance, the model is used to call attention to data outliers. For instance, banks detect fraud by looking for unusual purchase patterns with customers—like if a card is used in two very different locations in one day, the bank notices and investigates the activity.
**Artificial neural networks (or autoencoders):** An autoencoder takes input data, compresses it into code, and then tries to recreate the input from that code while removing any signal noise so data quality is improved.

source: [click here]( https://www.tableau.com/learn/articles/top-machine-learning-methods)



### 2.Ce este o retea neuronala (NN)?
 
Neural networks, also known as artificial neural networks (ANNs) or simulated neural networks (SNNs), are a subset of [machine learning](https://www.ibm.com/cloud/learn/machine-learning) and are at the heart of [deep learning](https://www.ibm.com/cloud/learn/deep-learning) algorithms. Their name and structure are inspired by the human brain, mimicking the way that biological neurons signal to one another.

Artificial neural networks (ANNs) are comprised of a node layers, containing an input layer, one or more hidden layers, and an output layer. Each node, or artificial neuron, connects to another and has an associated weight and threshold. If the output of any individual node is above the specified threshold value, that node is activated, sending data to the next layer of the network. Otherwise, no data is passed along to the next layer of the network.

![ICLH_Diagram_Batch_01_03-DeepNeuralNetwork-WHITEBG (1)](https://user-images.githubusercontent.com/97888771/185126979-504af1a0-0eeb-45f9-ab6c-609233fe5f7b.jpg)


Visual diagram of an input layer, hidden layers, and an output layer of a feedforward neural network
Neural networks rely on training data to learn and improve their accuracy over time. However, once these learning algorithms are fine-tuned for accuracy, they are powerful tools in computer science and [artificial intelligence](https://www.ibm.com/cloud/learn/what-is-artificial-intelligence), allowing us to classify and cluster data at a high velocity. Tasks in speech recognition or image recognition can take minutes versus hours when compared to the manual identification by human experts. One of the most well-known neural networks is Google’s search algorithm.

source : [click here](https://www.ibm.com/cloud/learn/neural-networks)

### 3 Ce reprezinta greutatea (weight)?

**Weight is the parameter within a [neural network](https://deepai.org/machine-learning-glossary-and-terms/neural-network) that transforms input data within the network's hidden layers. A neural network is a series of nodes, or [neurons](https://deepai.org/machine-learning-glossary-and-terms/neuron). Within each node is a set of inputs, weight, and a bias value**

![e20ff932-4269-4bed-82fb-383b0f1ce96d](https://user-images.githubusercontent.com/97888771/185138368-9f7353fc-1ede-4aa3-80e2-c8e20ac59607.png)


source : [click here](https://deepai.org/machine-learning-glossary-and-terms/weight-artificial-neural-network#:~:text=Weight%20is%20the%20parameter%20within,weight%2C%20and%20a%20bias%20value.)

### 4 Ce inseamna "strat neuronal" (layer)?
**A layer in a deep learning model is a structure or network topology in the architecture of the model, which take information from the previous layers and then pass information to the next layer.**

<img width="619" alt="HiddenLayers" src="https://user-images.githubusercontent.com/97888771/185136691-c48ec60c-9b59-465e-a925-d12540549d68.png">


source : [click here](https://en.wikipedia.org/wiki/Layer_(deep_learning))

### 5 Ce este o "epoca" (epoch)?

**An epoch in machine learning means one complete pass of the training dataset through the algorithm. This epochs number is an important hyperparameter for the algorithm. It specifies the number of epochs or complete passes of the entire training dataset passing through the training or learning process of the algorithm.**


source : [click here](https://www.jigsawacademy.com/blogs/ai-ml/epoch-in-machine-learning#:~:text=An%20epoch%20in%20machine%20learning,learning%20process%20of%20the%20algorithm.)


### 6 Ce reprezinta "functia de activare" (activation function)?

**An [activation function](https://en.wikipedia.org/wiki/Activation_function) in a neural network defines how the weighted sum of the input is transformed into an output from a node or nodes in a layer of the network.**

source : [click here](https://machinelearningmastery.com/choose-an-activation-function-for-deep-learning/#:~:text=An%20activation%20function%20in%20a,a%20layer%20of%20the%20network.)

### 7 Cum poate fi crescuta acuratetea modelului antrenat (accuracy)?

> - Add more data
> - Treat missing and Outlier values
> - Feature Engineering
> - Feature Selection
> - Multiple algorithms
> - Algorithm Tuning
> - Ensemble methods
> - Cross Validation


source : [click here](https://www.analyticsvidhya.com/blog/2015/12/improve-machine-learning-results/)


### 8 Ce sa-r schimba daca ar fi marit numarul de neuroni de sute de ori, la fel si numarul de straturi cu neuroni ?

==========================================================================


### 9 Cum mai este denumit trainingul / modelul cand se utilizeaza o retea cu mai multe straturi ?

**Stochastic Gradient Descent
The classical and still preferred training algorithm for neural networks is called stochastic gradient descent.

This is where one row of data is exposed to the network at a time as input. The network processes the input upward, activating neurons as it goes to finally produce an output value**

source : [click here](https://machinelearningmastery.com/neural-networks-crash-course/#:~:text=The%20classical%20and%20still%20preferred,finally%20produce%20an%20output%20value.)
